src/ispd/arquivo/xml/IconicoXML.java

```diff
@@ -82,1426 +26,809 @@
-                NodeList slaves = master.getElementsByTagName("slave");
-                CS_VMM mestre = (CS_VMM) centroDeServicos.get(global);
-                for (int j = 0; j < slaves.getLength(); j++) {
-                    Element slave = (Element) slaves.item(j);
-                    CentroServico maq = centroDeServicos.get(Integer.parseInt(slave.getAttribute("id")));
-                    if (maq instanceof CS_Processamento) {
-                        mestre.addEscravo((CS_Processamento) maq);
-                        if (maq instanceof CS_MaquinaCloud) {
-                            CS_MaquinaCloud maqTemp = (CS_MaquinaCloud) maq;
-                            //trecho de debbuging
-                            System.out.println(maqTemp.getId() + " adicionou como mestre: " + mestre.getId());
-                            //fim dbg
-                            maqTemp.addMestre(mestre);
-                        }
-                    } else if (maq instanceof CS_Switch) {
-                        for (CS_MaquinaCloud escr : escravosCluster.get(maq)) {
-                            escr.addMestre(mestre);
-                            mestre.addEscravo(escr);
-                        }
-                    }
-                }
-            }
-        }
-        
-        //Realiza leitura dos ícones de máquina virtual
-        for (int i = 0; i < docVMs.getLength(); i++) {
-            Element virtualMac = (Element) docVMs.item(i);
-            CS_VirtualMac VM = new CS_VirtualMac(virtualMac.getAttribute("id"),
-                    virtualMac.getAttribute("owner"),
-                    Integer.parseInt(virtualMac.getAttribute("power")),
-                    Double.parseDouble(virtualMac.getAttribute("mem_alloc")),
-                    Double.parseDouble(virtualMac.getAttribute("disk_alloc")),
-                    virtualMac.getAttribute("op_system"));
-            //adicionando VMM responsável pela VM
-            for (CS_Processamento aux : VMMs) {
-                //System.out.println("id vmm:" + aux.getId());
-                //System.out.println("id do vmm na vm:" + virtualMac.getAttribute("vmm") );
-                if (virtualMac.getAttribute("vmm").equals(aux.getId())) {
-                    //atentar ao fato de que a solução falha se o nome do vmm for alterado e não atualizado na tabela das vms
-                    //To do: corrigir problema futuramente
-                    VM.addVMM((CS_VMM) aux);
-                    //adicionando VM para o VMM
-                    
-                    CS_VMM vmm = (CS_VMM) aux;
-                    vmm.addVM(VM);
-                    
-                }
+    /**
+     * Add iconic model vertices and edges to the collections passed as
+     * arguments. <b>The collections are modified.</b>
+     *
+     * @param doc {@link Document} containing the iconic model
+     * @see IconicModelBuilder
+     */
+    public static void newGrade(
```

Comment:
The related generic type from the parameters _vertices_ and _edges_ **should not super** _Vertex_ and _Edge_, since, in that way would be possible do the following, create a list, `List<Object> list = new ArrayList<>();` that could previously contain any type of object and it could be passed as argument for both method parameters, that is, it would be correct the call `newGrade(doc, list, list)`; losing the desired semantic.